pkg/infra/game/stage/loot/loot.go

```diff
@@ -157,3 +166,99 @@ func itemListToSet(
 	}
 	return res
 }
+
+func getAverageStats(globalState state.State) (float64, float64, float64, float64) {
+	var averageHP float64 = 0
+	var averageST float64 = 0
+	var averageATT float64 = 0
+	var averageDEF float64 = 0
+
+	agentLen := float64(len(globalState.AgentState))
+	for _, state := range globalState.AgentState {
+		averageHP += float64(state.Hp)
+		averageST += float64(state.Stamina)
+		averageATT += float64(state.BonusAttack())
+		averageDEF += float64(state.BonusDefense())
+	}
+
+	averageHP /= agentLen
+	averageST /= agentLen
+	averageATT /= agentLen
+	averageDEF /= agentLen
+	// fmt.Println(averageHP, averageST, averageATT, averageDEF)
+	meanAverageHP, meanAverageST, meanAverageATT, meanAverageDEF := normalize4El(averageHP, averageST, averageATT, averageDEF)
+	// fmt.Println(meanAverageHP, meanAverageST, meanAverageATT, meanAverageDEF)
+	return meanAverageHP, meanAverageST, meanAverageATT, meanAverageDEF
+}
+
+func chooseItem(agent state.AgentState, averageHP float64, averageST float64, averageATT float64, averageDEF float64) (bool, bool, bool, bool) {
+
+	HP := float64(agent.Hp)
+	ST := float64(agent.Stamina)
+	ATT := float64(agent.BonusAttack())
+	DEF := float64(agent.BonusDefense())
+	// fmt.Println(HP, ST, ATT, DEF)
+	meanHP, meanST, meanATT, meanDEF := normalize4El(HP, ST, ATT, DEF)
+	// fmt.Println(meanHP, meanST, meanATT, meanDEF)
+	diffHP := averageHP - meanHP
+	diffST := averageST - meanST
+	diffATT := averageATT - meanATT
+	diffDEF := averageDEF - meanDEF
+	// fmt.Println(diffHP, diffST, diffATT, diffDEF)
+	// get largest diff = var most in need
+	if diffHP > diffST && diffHP > diffATT && diffHP > diffDEF {
+		return true, false, false, false // HP highest diff
+	} else if diffST > diffATT && diffST > diffDEF {
+		return false, true, false, false // ST highest diff
+	} else if diffATT > diffDEF {
+		return false, false, true, false // ATT highest diff
+	} else {
+		return false, false, false, true // DEF highest diff
+	}
```

Comment:
For speed/efficiency, you'd be better off returning a binary (1000, 0100 ...) and using bit operations, but that would require a large refactor